src/main/scala/com/fulcrumgenomics/vcf/filtration/EndRepairArtifactLikelihoodFilter.scala

```diff
@@ -0,0 +1,212 @@
+    val lnFractionIncongruent = LnProb.toLogProbability(refGood / (refGood + refBad).toDouble)
+    var llMutation = 1.0
+    var llArtifact = 1.0
+
+    // Now we can iterate over the alt bases and determine their likelihood under the two models
+    pileup.baseIterator.filter(_.base == altAllele).foreach { rec =>
+      val congruent = isArtifactCongruent(refAllele, altAllele, rec, pileup.pos)
+      val logProbError = (rec.qual / -10.0) * NumericTypes.LnTen // Convert Q into ln(pError)
+      val logProbRight = LnProb.not(logProbError)
+
+      if (congruent) {
+        altBad += 1
+        llArtifact += logProbRight
+        llMutation += LnProb.or(LnProb.and(logProbRight, lnFractionCongruent), LnProb.and(logProbError, lnFractionIncongruent))
+      }
+      else {
+        altGood += 1
+        llArtifact += logProbError
+        llMutation += LnProb.or(LnProb.and(logProbRight, lnFractionIncongruent), LnProb.and(logProbError, lnFractionCongruent))
+      }
+    }
+
+    // Apply a prior based on the MAF
+    val maf = (altGood + altBad) / (altGood + altBad + refGood + refBad).toDouble
+    val priorMutation = min(pow(maf * 2, 2), 1)
+    val priorArtifact = 1 - priorMutation
+    llMutation += LnProb.toLogProbability(priorMutation)
+    llArtifact += LnProb.toLogProbability(priorArtifact)
+
+    val total = LnProb.or(llMutation, llArtifact)
+    val pMutation = expProb(LnProb.normalizeByLogProbability(llMutation, total))
+    val pArtifact = expProb(LnProb.normalizeByLogProbability(llArtifact, total))
+
+    Map(InfoKey -> pMutation)
+  }
+
+  /** Calculates a pair of priors:
+    *   The first is the prior that the genotype is the result of a true somatic mutations
+    *   The second is the prior that the genotype is the result of the end-repair artifact process
+    *
+    * Default implementation gives a mutation prior of {{{(maf * 2)^2}}} capped at 0.9999 and an
+    * artifact prior of 1 - mutation prior.  This has the effect of creating priors that are heavily
+    * skewed towards the artifact at very low MAFs, and towards mutations as the MAF increases towards 0.5.
+    */
+  private[filtration] def priors(pileup: Pileup[PileupEntry], gt: Genotype, maf: Double) : (Double, Double) = {
+    require(maf > 0 && maf <= 1, "maf must be greater than 0 and no more than 1.")
+    val priorMutation = min(pow(maf * 2, 2), 0.9999)
+    val priorArtifact = 1 - priorMutation
+    (priorMutation, priorArtifact)
+  }
+
+  /** Exponentiates a log probability that should be in the range 0-1 and caps it there to account for numerical imprecision. */
+  private def expProb(lnP: LnProb): Double = {
+    val x = exp(lnP)
+    if (x > 1) 1 else if (x < 0) 0 else x
+  }
+
+  /**
+    * Returns true if a base in the pileup (either ref or alt) is congruent with the artifact.
+    * I.e. the base is within the first or last few bases of the template _and_ the base as read
+    * makes sense given the position.
```

Comment:
Agreed on not duplicating documentation, but could you link to the class with the explanation?